Parsers/ASimNetworkSession/Parsers/vimNetworkSessionCiscoFirepower.yaml

```diff
@@ -0,0 +1,318 @@
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let ActionLookup = datatable(
+      DeviceAction: string,
+      DvcAction: string,
+      EventResult: string
+  )
+  [
+      "Blocked", "Deny", "Failure",
+      "Alerted", "Allow", "Success",
+      "Rewritten", "Allow", "Success",
+      "Would be Rewritten", "Allow", "Partial",
+      "Would be Blocked", "Deny", "Partial",
+      "Would Be Blocked", "Deny", "Partial",
+      "Dropped", "Drop", "Failure",
+      "Would be Dropped", "Drop", "Partial",
+      "Partially Dropped", "Drop", "Partial",
+      "Would be Block", "Deny", "Partial",
+      "Partial Blocked", "Deny", "Partial",
+      "Rejected", "Deny", "Failure",
+      "Would be Rejected", "Deny", "Partial",
+      "Would Rejected", "Deny", "Partial",
+      "Block", "Deny", "Failure",
+      "Partial Block", "Deny", "Partial",
+      "Drop", "Drop", "Failure",
+      "Would Drop", "Drop", "Partial",
+      "Reject", "Deny", "Failure",
+      "Rewrite", "Allow", "Success",
+      "Allow", "Allow", "Success",
+      "Monitor", "Allow", "Success"
+  ];
+  let EventSeverityLookup = datatable (LogSeverity: string, EventSeverity: string)
+  [
+        "0", "Low",
+        "1", "Low",
+        "2", "Low",
+        "3", "Low",
+        "4", "Medium",
+        "5", "Medium",
+        "6", "Medium",
+        "7", "High",
+        "8", "High",
+        "9", "High",
+        "10", "High"
+  ];
+  let EventResultDetailsLookup = datatable(Reason: string, EventResultDetails: string)
+  [
+      "N/A", "NA",
+      "IP Block", "Terminated",
+      "IP Monitor", "Unknown",
+      "User Bypass", "Unknown",
+      "File Monitor", "Unknown",
+      "File Block", "Terminated",
+      "Intrusion Monitor", "Unknown",
+      "Intrusion Block", "Terminated",
+      "File Resume Block", "Terminated",
+      "File Resume Allow", "Unknown",
+      "File Custom Detection", "Unknown"
+  ];
+  let parser = (starttime: datetime=datetime(null), 
+      endtime: datetime=datetime(null),
+      srcipaddr_has_any_prefix: dynamic=dynamic([]), 
+      dstipaddr_has_any_prefix: dynamic=dynamic([]), 
+      ipaddr_has_any_prefix: dynamic=dynamic([]),
+      dstportnumber: int=int(null), 
+      hostname_has_any: dynamic=dynamic([]), 
+      dvcaction: dynamic=dynamic([]), 
+      eventresult: string='*', 
+      disabled: bool=false) {
+      let src_or_any = set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let dst_or_any = set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix);
+      let AllLogs = CommonSecurityLog
+          | where not(disabled)
+          | where (isnull(starttime) or TimeGenerated >= starttime) and (isnull(endtime) or TimeGenerated <= endtime)
+              and DeviceVendor == "Cisco" and DeviceProduct == "Firepower"
+              and DeviceEventClassID has_any("INTRUSION:400", "PV:112", "RNA:1003:1")
```

Comment:
@jayeshprajapaticrest @niralishah-crest 
Only these types of values have NetworkSession related logs? Also, can you please help sharing the logic behind categorizing it as an NetworkSession log.

Reply:
DeviceEventClassID has_any("INTRUSION:400", "PV:112", "RNA:1003:1")

Reply:
INTRUSION:400 = Intrusion Events
PV:112 = Policy Violation/ Correlation Events
RNA:1003:1 = Connection Statistics Events
By understanding meaning of each type of event and its fields we have categorized them as NetworkSession.
You can find details description about events in attached screenshots
![image](https://github.com/Azure/Azure-Sentinel/assets/107753570/549cdf8d-b4f3-4918-bded-ee1a535854b7)
![image](https://github.com/Azure/Azure-Sentinel/assets/107753570/469a7ff9-7eb0-4408-be91-543fa8a15280)